### ISSUE TITLE
hep buffer feature not working

### ISSUE THREAD

**Issue by hamidrhashmi** [sipcapture/heplify#297]
heplify version is 
```bash
/usr/local/bin/heplify -version
heplify 1.66.6
```
I am starting heplify with the following command
```bash
/usr/local/bin/heplify -hep-buffer-activate -hep-buffer-debug -hep-buffer-file /root/HEP.dump -e -l debug -hin tcp:0.0.0.0:9061 -hs debian10.*****.local:9062 -nt tcp -tcpsendretries 0 -tcpassembly -prometheus 0.0.0.0:8090
```
and as I stopped the hep server to test heplify buffering feature, it shows the following error
```
2024/07/30 11:08:15.774596 hep.go:173: ERR reconnect error: dial tcp 192.168.0.102:9062: connect: connection refused
2024/07/30 11:08:15.777539 hep.go:275: ERR adding packet to BUFFER: HEP3▒
▒▒
-▒
f▒ɟ

-

▒
 emptyhSIP/2.0 200 OK
Via: SIP/2.0/UDP 192.168.0.105:11656;rport=11656;received=192.168.0.105;branch=z9hG4bK-524287-1---da45306c67efa84f
Call-ID: ICoelyS2IlqjAeH2GpdQ7w..
From: "THA***********" <sip:THA923226363266@192.168.0.133>;tag=2d36e97c
To: <sip:THA***********@192.168.0.133>;tag=316622b442d14b069ed4f5f1f8f2c2c9
CSeq: 3 BYE
Content-Length:  0
```
and if I check the buffer file it only contains one packet
```
 cat /root/HEP.dump
HEP3▒
▒▒
▒▒
f▒▒^

]

▒
 empty▒BYE sip:THA923009682285@192.168.0.106:7060;ob SIP/2.0
Via: SIP/2.0/UDP 192.168.0.133;branch=z9hG4bK5c0e.664b72e6000000000000000000000000.0
To: <sip:THA***********@kamailio.*****.local>;tag=af6f8130386f49d988f681ba4a7a9dad
From: <sip:THA***********@kamailio.*****.local>;tag=b93bcc5c
CSeq: 1 BYE
Call-ID: ICoelyS2IlqjAeH2GpdQ7w..
Content-Length: 0
Max-Forwards: 70
```

and on reconnecting (turning hep server back up) it didn't flush up the file.

**Comment by adubovikov** [comment]
please check 

https://github.com/sipcapture/heplify/pull/new/fix_buffer

**Comment by hamidrhashmi** [comment]
@adubovikov Its still giving the same error 
```
2024/08/12 10:08:24.430726 hep.go:172: ERR reconnect error: dial tcp 10.124.0.4:9062: connect: connection refused
2024/08/12 10:08:24.430742 hep.go:272: ERR adding packet to BUFFER: HEP3▒
▒▒
#▒
f▒▒

▒

▒
 empty▒SIP/2.0 200 OK
Via: SIP/2.0/UDP 192.168.0.184:9117;rport=9117;received=192.168.0.184;branch=z9hG4bK-524287-1---17a0527a679eac0f
Call-ID: a741b643bba4462194df71702a2f6702
From: <sip:THA923226363266@kamailio.hbvoice.local>;tag=ff58e053
To: "Hamid Raza Hashmi" <sip:THA923009682285@kamailio.hbvoice.local>;tag=6977368237c74769903e6c1199837f74
CSeq: 2 BYE
Content-Length:  0

2024/08/12 10:08:24.430805 hep.go:302: DBG [collector]  File Send HEP from buffer to file OK
```

Please find the complete logs in the attached file 
[heplify_hep.txt](https://github.com/user-attachments/files/16580220/heplify_hep.txt)

**Comment by hamidrhashmi** [comment]
Its been fixed in the latest version 1.66.7
